src/cdk-experimental/column-resize/overlay-handle.ts

```diff
@@ -0,0 +1,195 @@
+
+        mousemove.pipe(
+            takeUntil(mouseup),
+            takeUntil(escape),
+            takeUntilDestroyed,
+            map(({pageX}) => pageX),
+            startWith(startX),
+            distinctUntilChanged(),
+            pairwise(),
+        ).subscribe(([prevX, currX]) => {
+          let deltaX = currX - prevX;
+
+          if (overshot !== 0) {
+            if (overshot < 0 && deltaX < 0 || overshot > 0 && deltaX > 0) {
+              overshot += deltaX;
+              return;
+            } else {
+              const remainingOvershot = overshot + deltaX;
+              overshot = overshot > 0 ?
+                  Math.max(remainingOvershot, 0) : Math.min(remainingOvershot, 0);
+              deltaX = remainingOvershot - overshot;
+
+              if (deltaX === 0) {
+                return;
+              }
+            }
+          }
+
+          let computedNewSize: number = size + (this._isLtr() ? deltaX : -deltaX);
+          computedNewSize = Math.min(
+              Math.max(computedNewSize, this.resizeRef.minPx, 0), this.resizeRef.maxPx);
+
+          this.resizeNotifier.triggerResize.next({columnId, size: computedNewSize});
+
+          const originNewSize = this._getOriginWidth();
+          const originNewOffset = this._getOriginOffset();
+          const originOffsetDeltaX = originNewOffset - originOffset;
+          const originSizeDeltaX = originNewSize - size;
+          size = originNewSize;
+          originOffset = originNewOffset;
+
+          overshot += deltaX + (this._isLtr() ? -originSizeDeltaX : originSizeDeltaX);
+          overlayOffset += originSizeDeltaX + originOffsetDeltaX;
+
+          this._updateOverlayOffset(overlayOffset);
+        });
+      });
+    });
+  }
+
+  protected updateResizeActive(active: boolean): void {
+    this.eventDispatcher.overlayHandleActiveForCell.next(
+        active ? this.resizeRef.origin.nativeElement! : null);
+  }
+
+  private _getOriginWidth(): number {
+    return this.resizeRef.origin.nativeElement!.offsetWidth;
+  }
+
+  private _getOriginOffset(): number {
+    const originElement = this.resizeRef.origin.nativeElement!;
+    const offsetLeft = originElement.offsetLeft;
+
+    return this._isLtr() ?
+        offsetLeft :
+        originElement.offsetParent!.offsetWidth - (offsetLeft + this._getOriginWidth());
+  }
+
+  private _getOverlayOffset(): number {
+    const overlayElement = this.resizeRef.overlayRef.overlayElement;
+    return this._isLtr() ?
+        parseInt(overlayElement.style.left!, 10) : parseInt(overlayElement.style.right!, 10);
+  }
+
+  private _updateOverlayOffset(offset: number): void {
+    const overlayElement = this.resizeRef.overlayRef.overlayElement;
+    const overlayOffsetCssValue = coerceCssPixelValue(offset);
+
+    if (this._isLtr()) {
+      overlayElement.style.left = overlayOffsetCssValue;
+    } else {
+      overlayElement.style.right = overlayOffsetCssValue;
+    }
+  }
+
+  private _isLtr(): boolean {
```

Comment:
optional: we usually use `isRtl` throughout the library (though tooltip does also use isLtr)